test/runtests.jl

```diff
@@ -3,6 +3,31 @@ using Legolas, Test, DataFrames, Arrow, UUIDs
 using Legolas: SchemaVersion, @schema, @version, SchemaVersionDeclarationError, DeclaredFieldInfo
 using Accessors
 using Aqua
+using Pkg
+
+# This test set goes before we load `TestProviderPkg`
+@testset "#46: Informative errors when reading unknown schemas from packages" begin
+    err = Legolas.UnknownSchemaVersionError(Legolas.SchemaVersion("test-provider-pkg.foo", 1), :TestProviderPkg, v"0.1.0")
+    @test_throws err Legolas.read("test_provider_pkg.arrow")
+    @test contains(sprint(Base.showerror, err), "TestProviderPkg")
+
+    # Let's test some more error printing while we're here; if we did not have the VersionNumber
+    # (e.g. since the table was generated on Julia pre-1.9), we should still print a reasonable message:
+    err = Legolas.UnknownSchemaVersionError(Legolas.SchemaVersion("test-provider-pkg.foo", 1), :TestProviderPkg, missing)
+    @test contains(sprint(Base.showerror, err), "TestProviderPkg")
```

Comment:
We should use `Arrow.Table` to load the table without the package to ensure that the advice we give is accurate